### PR TITLE
Fill OSError attributes

### PR DESCRIPTION
**Pull request by dvermd**
Tested by removing expected failure and commenting everything but OSError in Lib/test/test_exceptions.py::testAttributes::exceptionList, WindowsError and pickling part of the test.

fixes : #4193

### CODE REVIEW

**Comment by dvermd** [comment]
From the discussion in #4191, there seems to be 2 different behaviors of the OSError constructor:

- the one which takes from 2 to 5 arguments and that fills errno, strerror, winerror, filename and filename2
- the other one which takes 1 or more than 5 arguments, that just behaves like a BaseException without filling any of the OSError fields

File: Lib/test/test_exceptions.py [hunk 1]
```diff
@@ -410,6 +410,32 @@ def test_WindowsError(self):
...
+            self.assertEqual(w.filename2, None)
+            self.assertEqual(str(w), "foo")
+
+            w = OSError('a', 'b', 'c', 'd', 'e', 'f')
+            self.assertEqual(w.errno, None)
+            self.assertEqual(w.winerror, None)
+            self.assertEqual(w.strerror, None)
+            self.assertEqual(w.filename, None)
+            self.assertEqual(w.filename2, None)
+            self.assertEqual(str(w), "('a', 'b', 'c', 'd', 'e', 'f')")
```
Comment (youknowone) [inline Lib/test/test_exceptions.py:438]: if this our own test or copied from cpython? we usually don't write our own tests here but in `extra_tests/snippets/`

Comment (dvermd) [reply Lib/test/test_exceptions.py]: Moved the new tests to `extra_tests/snippets/builtin_exceptions.py`

**Review by youknowone — APPROVED** [review]
thank you!

**Pull request merged**